_rules/presentational-children-no-focusable-content-307n5z.md

```diff
@@ -0,0 +1,144 @@
+
+## Assumptions
+
+_There are currently no assumptions_
+
+## Accessibility Support
+
+Several major browsers ignore the WAI-ARIA requirements on [presentational children][] for most or sometimes all roles. Because some browsers do, and others do not implement presentational children, there can be big differences between browsers.
+
+## Background
+
+This rule applies to elements with a [semantic role][] that defines its [children][child] to be [presentational children][]. This includes the following: `button`, `checkbox`, `img`, `math`, `menuitemcheckbox`, `menuitemradio`, `option`, `progressbar`, `radio`, `scrollbar`, `separator`, `slider`, `switch`, and `tab`.
+
+Elements with a [semantic role][] that has [presentational children][] will not have any descendants in the accessibility tree. If any of those descendants are included in [sequential focus navigation][], this causes the focus to land on an element that has no corresponding node in the [accessibility tree][]. The result is that there is no programmatic name or role available for assistive technologies. There are other problems that can come from [presentational children][] too. These must be tested separately.
+
+- [Understanding Success Criterion 1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships)
+- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)
+- [WAI-ARIA 1.1 Presentational Children][presentational children]
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+None of these `button` elements has [descendants][] that are included in [sequential focus navigation][].
+
+```html
+<button>Save</button> <button aria-label="save options" aria-expanded="false">▼</button>
+```
+
+#### Passed Example 2
+
+This element with `checkbox` role has no [descendants][] that are included in [sequential focus navigation][]. Instead the link to the terms of service is adjacent, and `aria-labelledby` is used to provide its [accessible name][].
+
+```html
+<p id="terms">
+	<span role="checkbox" aria-checked="false" tabindex="0" aria-labelledby="terms">
+		I agree to the
+	</span>
+	<a href="/terms">terms of service</a>
+</p>
+```
+
+#### Passed Example 3
+
+This element with `menuitemcheckbox` role has an `input` element as a descendant. Because the `input` is disabled it is not included in [sequential focus navigation][].
+
+**Note**: The `input` checkbox has a `role` [attribute value][] of `none` to ensure it is ignored by browsers that do not support [presentational children][].
+
+```html
+<ul role="menu">
+	<li role="menuitemcheckbox" aria-checked="true">
+		<input type="checkbox" role="none" disabled checked />
+		Sort by Last Modified
+	</li>
+</ul>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This `button` element has a [child][] `span` element that is included in [sequential focus navigation][].
```

Comment:
```suggestion
This `button` element has a [child][] `span` element. Because the `span` element has a `tabindex` attribute with value of `0`, it is included in [sequential focus navigation][].
```
I think it is worth explaining why in the description.